File: sudoku-common/common-alicloud/src/main/java/com/zumin/sudoku/common/alicloud/config/OSSConfig.java
package com.zumin.sudoku.common.alicloud.config;

import com.aliyun.oss.OSS;
import com.aliyun.oss.OSSClientBuilder;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

/**
 * OSS配置类
 */
@Configuration
public class OSSConfig {

  @Value("${alicloud.access-key-id}")
  private String accessKeyId;
  @Value("${alicloud.secret-key-secret}")
  private String secretKeySecret;
  @Value("${alicloud.oss.endpoint}")
  private String endpoint;

  /**
   * 实例化一个OSS对象到容器中
   *
   * @return OSS对象
   */
  @Bean
  public OSS oss() {
    return new OSSClientBuilder().build(endpoint, accessKeyId, secretKeySecret);
  }
}

File: sudoku-common/common-alicloud/src/main/java/com/zumin/sudoku/common/alicloud/OSSConfig.kt
package com.zumin.sudoku.common.alicloud

import com.aliyun.oss.OSS
import com.aliyun.oss.OSSClientBuilder
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

/**
 * OSS配置类
 */
@Configuration
@EnableConfigurationProperties(AliCloudProperties::class)
class OSSConfig(val aliCloudProperties: AliCloudProperties) {

  /**
   * 实例化一个OSS对象到容器中
   *
   * @return OSS对象
   */
  @Bean
  fun oss(): OSS {
    val (accessKeyId, secretKeySecret, oss) = aliCloudProperties
    return OSSClientBuilder().build(oss.endpoint, accessKeyId, secretKeySecret)
  }
}